src/toil/cwl/cwltoil.py

```diff
@@ -1208,6 +1333,8 @@ def toil_get_file(
     to close the file descriptors (to break the pipes) and join the threads
     """
     pipe_threads_real = pipe_threads or []
+    # We can't use urlparse here because we need to handle the '_:' scheme and
+    # urlparse sees that as a path and not a URI scheme.
```

Comment:
Hm. `urlparse` does in fact have a long list of specific URL schemes it supports, instead of just saying that it parses URLs with a general kind of scheme.

I wanted to use `urlparse` because it, mostly, supports telling the difference between something with a scheme and something without. I want to be able to say `if scheme in ['file', '']` to catch the file URIs and the bare paths. Right now that works as long as I've already pulled out `_:` in another case.

Maybe we need our own URI parsing function at some point?